### PR TITLE
[subgraphDumper] Setup op priority based on opset version

### PR DESCRIPTION
**Pull request by sbalandi**
### Details:
 - *update setting of priority for models, now it 2 - if it is used latest version of operation, 1 - in all other cases*
 - *function get_unique_ops() moved to functionalTestUtils*
 - *updated test update_cache_by_model and added tests get_max_ops_versions and  get_node_priority_by_version to NodeUtilsTest*

### Tickets:
 - *CVS-105739*

### CODE REVIEW

**Comment by iefode** [comment]
Please provide useful description :)

File: src/tests/test_utils/common_test_utils/include/common_test_utils/common_utils.hpp [hunk 1]
```diff
@@ -186,4 +187,26 @@ inline std::ostream& operator<<(std::ostream& os, const std::map<std::string, st
 }
 
 std::string generateTestFilePrefix();
+
+static std::map<std::string, std::set<std::string>> get_unique_ops() {
```
[the remark this answers is not in the document]
Comment (sbalandi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/common_utils.hpp]: comment is added, function is moved to note_utils.hpp

File: src/tests/test_utils/common_test_utils/include/common_test_utils/common_utils.hpp [hunk 2]
```diff
@@ -186,4 +187,26 @@ inline std::ostream& operator<<(std::ostream& os, const std::map<std::string, st
...
+    for (const auto& opset_pair : ov::get_available_opsets()) {
+        std::string opset_name = opset_pair.first;
+        const ov::OpSet& opset = opset_pair.second();
+        for (const auto& op : opset.get_type_info_set()) {
+            std::string op_version = op.get_version(), opset_name = "opset";
+            auto pos = op_version.find(opset_name);
+            if (pos != std::string::npos) {
+                op_version = op_version.substr(pos + opset_name.size());
+            }
+            if (res.find(op.name) == res.end()) {
```
[the remark this answers is not in the document]
Comment (sbalandi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/common_utils.hpp]: removed

File: src/tests/functional/plugin/conformance/subgraphs_dumper_new/tests/cache/op_cache.cpp [hunk 3]
```diff
@@ -105,17 +105,17 @@ TEST_F(OpCacheUnitTest, update_cache_by_model) {
         param->set_friendly_name("in_0");
```
[the remark this answers is not in the document]
Comment (sbalandi) [reply src/tests/functional/plugin/conformance/subgraphs_dumper_new/tests/cache/op_cache.cpp:105]: tests are added

File: src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp [hunk 4]
```diff
@@ -167,6 +168,48 @@ inline std::string get_node_type(const std::shared_ptr<ov::Node>& node) {
...
+        }
+        int max_opset = *std::max_element(available_opsets.begin(), available_opsets.end());
+
+        max_ops_versions.insert({op_info.first, std::to_string(max_opset)});
+    }
+    return max_ops_versions;
+}
+
+//  { op_name, max_opset_version }
+static std::map<std::string, std::string> max_ops_versions = ov::tools::subgraph_dumper::get_max_ops_versions();
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp:189]: we can move this definition inside `get_node_priority_by_version()`. Looks like we did not use it in other places

File: src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp [hunk 5]
```diff
@@ -167,6 +168,48 @@ inline std::string get_node_type(const std::shared_ptr<ov::Node>& node) {
...
+
+inline size_t get_node_priority_by_version(const std::shared_ptr<ov::Node>& node) {
+    size_t priority = 1;
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp:192]:
```
    auto max_opset_it = std::max_element
(
    max_ops_versions.begin(), max_ops_versions.end(),
    [] (std::pair<std::string, std::string>& orig, [const pair_type](std::pair<std::string, std::string>)& ref) {
        return orig.second <ref.second;
    }
);
auto max_opset_version = max_opset_it.second;
```

File: src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp [hunk 6]
```diff
@@ -167,6 +168,48 @@ inline std::string get_node_type(const std::shared_ptr<ov::Node>& node) {
...
+        std::string version(type_info.version_id);
+        auto pos = version.find(opset_name);
+        if (pos != std::string::npos) {
+            version_id = version.substr(pos + opset_name.size());
+        }
+
+        if (version_id == ov::tools::subgraph_dumper::max_ops_versions[type_info.name]) {
+            priority = 2;
+        }
+    }
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp:208]:
```
if (version_id == max_opset_version) {
    priority = 3;
}
```

File: src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp [hunk 7]
```diff
@@ -167,6 +168,48 @@ inline std::string get_node_type(const std::shared_ptr<ov::Node>& node) {
...
+    std::map<std::string, std::set<std::string>> unique_ops = FuncTestUtils::get_unique_ops();
+    std::map<std::string, std::string> max_ops_versions;
+
+    size_t priority = 1;
+    for (auto op_info : unique_ops) {
+        std::vector<int> available_opsets;
+        for (auto opset : op_info.second) {
+            available_opsets.push_back(std::atoi(opset.c_str()));
+        }
+        int max_opset = *std::max_element(available_opsets.begin(), available_opsets.end());
```
Comment (iefode) [inline src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp:181]: std::string:
`opset2` > `opset12`

it is incorrect in our case!the simplest way is to extract opset number and convert to size_t

Comment (sbalandi) [reply src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp:66]: I don't got it, I compare versions as int here, but as string it is right way ?

File: src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp (hunk 6 above)
Comment (sbalandi) [reply src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp:101]: added

File: src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp (hunk 4 above)
Comment (sbalandi) [reply src/tests/functional/plugin/conformance/subgraphs_dumper_new/include/utils/node.hpp]: moved

**Review by iefode — APPROVED** [review]

**Pull request merged**